src/Confluent.Kafka/Serializers.cs

```diff
@@ -0,0 +1,196 @@
+        /// </returns>
+        public static byte[] Int32(string topic, int data)
+        {
+            var result = new byte[4]; // int is always 32 bits on .NET.
+            // network byte order -> big endian -> most significant byte in the smallest address.
+            // Note: At the IL level, the conv.u1 operator is used to cast int to byte which truncates
+            // the high order bits if overflow occurs.
+            // https://msdn.microsoft.com/en-us/library/system.reflection.emit.opcodes.conv_u1.aspx
+            result[0] = (byte)(data >> 24);
+            result[1] = (byte)(data >> 16); // & 0xff;
+            result[2] = (byte)(data >> 8); // & 0xff;
+            result[3] = (byte)data; // & 0xff;
+            return result;
+        }
+
+        /// <summary>
+        ///     Serializes the specified System.Single value to a byte array of length 4. Byte order is big endian (network byte order).
+        /// </summary>
+        /// <param name="topic">
+        ///     The topic associated with the data (ignored by this serializer).
+        /// </param>
+        /// <param name="data">
+        ///     The System.Single value to serialize.
+        /// </param>
+        /// <returns>
+        ///     The System.Single value <paramref name="data" /> encoded as a byte array of length 4 (network byte order).
+        /// </returns>
+        public static byte[] Float(string topic, float data)
+        {
+            if (BitConverter.IsLittleEndian)
+            {
+                unsafe
+                {
+                    byte[] result = new byte[4];
+                    byte* p = (byte*)(&data);
+                    result[3] = *p++;
+                    result[2] = *p++;
+                    result[1] = *p++;
+                    result[0] = *p++;
+                    return result;
+                }
+            }
+            else
+            {
+                return BitConverter.GetBytes(data);
+            }
+        }
+
+        /// <summary>
+        ///     Serializes the specified System.Double value to a byte array of length 8. Byte order is big endian (network byte order).
+        /// </summary>
+        /// <param name="topic">
+        ///     The topic associated with the data (ignored by this serializer).
+        /// </param>
+        /// <param name="data">
+        ///     The System.Double value to serialize.
+        /// </param>
+        /// <returns>
+        ///     The System.Double value <paramref name="data" /> encoded as a byte array of length 4 (network byte order).
+        /// </returns>
+        public static byte[] Double(string topic, double data)
+        {
+            if (BitConverter.IsLittleEndian)
+            {
+                unsafe
+                {
+                    byte[] result = new byte[8];
+                    byte* p = (byte*)(&data);
+                    result[7] = *p++;
```

Comment:
```

      if (BitConverter.IsLittleEndian)
         Array.Reverse(bytes);
```

Reply:
would be less efficient (though less error prone).